src/FastConv/FastConv.cpp

```diff
@@ -10,23 +11,259 @@ CFastConv::~CFastConv( void )
+            sum += m_pfInputBlock[m] * m_pfIRMatrix[idx][i - m];
+        }
+        m_pfBlockConvOuput[i] = sum;
+    }
+    return Error_t::kNoError;
+}
+
+
+Error_t CFastConv::freqConv(int idx)
+{
+    m_pfTimeIr = m_pfIRMatrix[idx];
+    // Take FFT of x and h, store in m_pfFreqInput, m_pfFreqIr
+    m_pCFftInstance->doFft(m_pfFreqInput, m_pfTimeInput);
+    m_pCFftInstance->doFft(m_pfFreqIr, m_pfIRMatrix[idx]);
+    // Split the real and imaginary parts for multiplication
+    m_pCFftInstance->splitRealImag(m_pfRealInput, m_pfImagInput, m_pfFreqInput);
+    m_pCFftInstance->splitRealImag(m_pfRealIr, m_pfImagIr, m_pfFreqIr);
+
+    // frequency domain multiplication == time domain convolution
+    for (int i = 0; i < (2 * m_iBlockLength); i++)
+    {
+        float ac =  m_pfRealInput[i] * m_pfRealIr[i];
+        float bc = m_pfImagInput[i] * m_pfRealIr[i];
+        float ad = m_pfRealInput[i] * m_pfRealIr[i];
+        float bd = m_pfImagInput[i] * m_pfImagIr[i];
+        m_pfRealConv[i] = ac - bd;
+        m_pfImagConv[i] = ad + bc;
+    }
+    m_pCFftInstance->mergeRealImag(m_pfFreqConv, m_pfRealConv, m_pfImagConv);
+    // Convert back to time domain
+    m_pCFftInstance->doInvFft(m_pfConvOuput, m_pfFreqConv);
+}
+
+
+
+
+Error_t CFastConv::conv()
+{
+    //clear previous values in m_pfConvMatrix
+    for (int i=0; i<m_iBlockLength*(m_iBlockNum+1); i++)
+        m_pfConvMatrix[m_iCirIdx][i] = 0.f;
+    
+    for (int i=0; i<m_iBlockLength; i++)
+        m_pfOutputBlock[i] = 0.f;
+    
+    for (int i=0; i<m_iBlockLength*2; i++)
+        m_pfBlockConvOuput[i] = 0.f;
+    
+    
+    for (int i = 0; i<m_iBlockNum; i++)
+    {
+        //switch to freqConv() after implemented
+        switch (m_eCompMode) {
+            case kTimeDomain:
+                timeConv(i);
+                break;
+            case kFreqDomain:
+                freqConv(i);
+                break;
+            case kNumConvCompModes:
+                break;
+        }
+        
+        
+        for (int n = 0; n < m_iBlockLength*2 - 1; n++)
+            m_pfConvMatrix[m_iCirIdx][n+i*m_iBlockLength] += m_pfBlockConvOuput[n];
+    }
+    
+    for (int i = 0; i < m_iBlockNum+1; i++)
+    {
+        int iMatrixIdx = (m_iCirIdx+m_iBlockNum+1-i) % (m_iBlockNum+1);
+        for (int j = 0; j < m_iBlockLength; j++)
+            m_pfOutputBlock[j] += m_pfConvMatrix[iMatrixIdx][i*m_iBlockLength+j];
+    }
+    
     return Error_t::kNoError;
 }
+ 
 
-Error_t CFastConv::process (float* pfOutputBuffer, const float *pfInputBuffer, int iLengthOfBuffers )
+Error_t CFastConv::process (float* pfOutputBuffer, const float *pfInputBuffer, int iLengthOfBuffers)
 {
+    if (!m_bIsInitialized)
+    {
+        std::cout<< "not initialized";
+        return Error_t::kNotInitializedError;
+    }
+    
+    
+    m_iReset = 0;
+    
+    for (int i=0; i<iLengthOfBuffers; i++)
+    {
+        m_pfInputBlock[i-m_iReset] = pfInputBuffer[i];
+
+        if ((i+1) % m_iBlockLength == 0)
```

Comment:
Note that you still need to write to the output buffer even if `process` is called with less than `m_iBlockLength` samples. (This code doesn't, which is probably why your test `timedomainblocksize` fails.)